File: group1/frontend/src/request/LetterStatus.ts
/**
 * Representation of every possible keyboard letter
 */
export type Letter =
    'a' | 'b' | 'c' | 'd' | 'e' | 'f' | 'g' | 'h' | 'i' | 'j' | 'k' | 'l' | 'm' |
    'n' | 'o' | 'p' | 'q' | 'r' | 's' | 't' | 'u' | 'v' | 'w' | 'x' | 'y' | 'z' |
    'A' | 'B' | 'C' | 'D' | 'E' | 'F' | 'G' | 'H' | 'I' | 'J' | 'K' | 'L' | 'M' |
    'N' | 'O' | 'P' | 'Q' | 'R' | 'S' | 'T' | 'U' | 'V' | 'W' | 'X' | 'Y' | 'Z';

/**
 * Represents the accuracy of a specific letter in a word
 */
export enum Accuracy {
    Wrong,
    Misplaced,
    Correct,
}

/**
 * Contains all the information about a letter in a word
 */
export class LetterStatus {
    public letter: Letter;
    public accuracy: Accuracy;

    constructor(letter: Letter, accuracy: Accuracy) {
        this.letter = letter;
        this.accuracy = accuracy;
    }

    public static fromAxiosResponse(response: any): LetterStatus {
        if (response.accuracy === "CORRECT"){
            return new LetterStatus(response.letter, Accuracy.Correct);
        } else if (response.accuracy === "MISPLACED"){
            return new LetterStatus(response.letter, Accuracy.Misplaced);
        } else if (response.accuracy === "WRONG"){
            return new LetterStatus(response.letter, Accuracy.Wrong);
        }
        throw new Error("Invalid accuracy: " + response.accuracy);
    }

}

File: group1/backend/src/main/java/com/group1/backend/game/LetterStatus.java
package com.group1.backend.game;

/**`
 * Represents the status of a letter in a word
 */
public class LetterStatus {
    private final char letter;
    private final Accuracy accuracy;

    /**
     * Create a new dummy LetterStatus
     * @deprecated Use the constructor that takes a letter and accuracy
     */
    @Deprecated
    public static LetterStatus dummyStatus() {
        return new LetterStatus('h', Accuracy.WRONG);
    }

    /**
     * Create a new LetterStatus with the given letter and accuracy
     * @param letter The letter
     * @param accuracy The accuracy
     */
    public LetterStatus(char letter, Accuracy accuracy) {
        this.letter = letter;
        this.accuracy = accuracy;
    }

    /**
     * Get the letter
     * @return The letter
     */
    public char getLetter() {
        return letter;
    }

    /**
     * Get the accuracy
     * @return The accuracy
     */
    public Accuracy getAccuracy() {
        return accuracy;
    }

    /**
     * Represents the accuracy of a letter
     */
    public enum Accuracy {
        WRONG,
        MISPLACED,
        CORRECT,
    }

    @Override
    public String toString() {
        if(accuracy == Accuracy.WRONG){
            return "WRONG";
        }else if(accuracy == Accuracy.MISPLACED){
            return "MISPLACED";
        }else if(accuracy == Accuracy.CORRECT){
            return "CORRECT";
        }

        return "Unassigned";
    }
}
